### PR TITLE
Integration Tests: set up `tests/integration` with success path 

### LINKED ISSUE #160: [CI/CD] Add integration tests and set up in Github to run in PR
Best have 3 databases for different environments:
test_db
dev_db
prod_db

Then integration tests can create whatever test records in test_db and verify functionality during PR merge

### PR DESCRIPTION
**Pull request by ginaxu1**
## Summary
**THIS IS PART 1, will add more integration tests for failure path and edge cases in the subsequent PRs** 
This PR deletes all scripts (outdated) and adds a Go test `graphql_flow_test.go` that validates the complete success path through the data exchange platform, ensuring proper communication between OE, PDP and CE. 

The test covers: 
- GraphQL query to orchestration-engine-go
- policy evaluation via PDP
- consent verification via consent-engine
- response validation

The `docker-compose.test.yml` now includes all three services (orchestration-engine, policy-decision-point, consent-engine) with their PostgreSQL databases (pdp-db on port 5433, ce-db on port 5434) on a shared test network

## Related Issues
Closes #160

### CODE REVIEW

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+on:
+  # Run on pushes to dev (for deployment)
+  push:
+    branches:
+      - dev
+  # Run on pull requests to dev (for testing)
+  pull_request:
+    branches:
+      - dev
```
Comment (sthanikan2000) [inline .github/workflows/integration-tests.yml]:
```suggestion
      - main
```

File: .github/workflows/integration-tests.yml [hunk 2]
```diff
@@ -0,0 +1,270 @@
+name: OpenDIF CI/CD Pipeline
+
+on:
+  # Run on pushes to main (for deployment)
+  push:
+    branches:
+      - main
```
Comment (sthanikan2000) [inline .github/workflows/integration-tests.yml]: Is it need to check on the push to main? It's already checking on the PR to main right?

Comment (ginaxu1) [reply .github/workflows/integration-tests.yml]: for safety, we should keep this double run. here is why: 

```
 # Run on pull requests to main (for testing before merge)
  pull_request:
    branches:
      - main
  # Run on pushes to main (for deployment jobs - deploy_and_smoke_test and build_and_push_images)
  # Note: build_and_test will run again on push, but this is intentional for:
  # 1. Direct pushes to main (bypassing PRs)
  # 2. Safety net in case PR tests were skipped
  push:
    branches:
      - main
```

File: .github/workflows/policy-decision-point-validate.yml
Comment (sthanikan2000) [inline .github/workflows/policy-decision-point-validate.yml]: Do the same change in the `.github/workflows/policy-decision-point-publish.yml` as well

File: exchange/orchestration-engine/Dockerfile [hunk 3]
```diff
@@ -8,13 +8,16 @@ RUN apk add --no-cache git ca-certificates
...
+
+# Copy shared packages (if needed, though OE might not use it yet, better safe)
+# COPY shared/ /shared/ 
 
 # Download dependencies
 RUN go mod download
 
 # Copy source code
-COPY . .
+COPY orchestration-engine/ .
```
Comment (sthanikan2000) [inline exchange/orchestration-engine/Dockerfile]: I can see this is different from the docker files changes done at `exchange/consent-engine/Dockerfile` and the `exchange/policy-decision-point/Dockerfile`, Is there any specific reason for this is different from the other files?
Can't we have the same logics to maintain the consistency across the files?
```suggestion
# Copy go mod files and source code
COPY orchestration-engine/go.mod orchestration-engine/go.sum ./
COPY orchestration-engine/ .

# Copy shared packages
COPY shared/ /shared/ 

# Download dependencies
RUN go mod download
```

File: .github/workflows/integration-tests.yml [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+  # Run on pull requests to main (for testing)
+  pull_request:
+    branches:
+      - main
```
[the remark this answers is not in the document]
Comment (ginaxu1) [reply .github/workflows/integration-tests.yml]: updated per your suggestion
